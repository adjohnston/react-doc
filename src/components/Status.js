import React from 'react'
import PropTypes from 'prop-types'
import classString from 'helpers/class-string'
import constants from 'constants/status'

const Status = ({status}) => ((!!status && (
  <span
    className={`${classString('__status')} ${classString(`__status--${constants[status].className}`)}`}>
    {constants[status].text}
  </span>
)) || null)

Status.propTypes = {
  status: PropTypes.oneOf(
    ['DANGEROUS', 'WIP', 'READY']
  )
}

export default Status
