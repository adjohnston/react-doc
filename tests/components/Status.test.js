import React from 'react'
import renderer from 'react-test-renderer'
import Status from 'src/components/Status'

test('it should render correctly with status - DANGEROUS', () => {
  const mockProps = {
    status: 'DANGEROUS'
  }

  const tree = renderer
    .create(<Status {...mockProps} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

test('it should render correctly with status - WIP', () => {
  const mockProps = {
    status: 'WIP'
  }

  const tree = renderer
    .create(<Status {...mockProps} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

test('it should render correctly with status - READY', () => {
  const mockProps = {
    status: 'READY'
  }

  const tree = renderer
    .create(<Status {...mockProps} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

test('it should render null without a status', () => {
  const tree = renderer
    .create(<Status />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
