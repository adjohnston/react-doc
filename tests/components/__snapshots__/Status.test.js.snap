// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render correctly with status - DANGEROUS 1`] = `
<span
  className="stncl__status stncl__status--dangerous"
>
  Dangerous
</span>
`;

exports[`it should render correctly with status - READY 1`] = `
<span
  className="stncl__status stncl__status--ready"
>
  Ready
</span>
`;

exports[`it should render correctly with status - WIP 1`] = `
<span
  className="stncl__status stncl__status--wip"
>
  Work in progress
</span>
`;

exports[`it should render null without a status 1`] = `null`;
